lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/ts_shape_inference.h

```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+#include "lazy_tensor_core/csrc/ts_backend/ts_lowering_context.h"
+
+namespace torch_lazy_tensors {
+namespace compiler {
+/**
+ * This is a deprecated interface, to be replaced by LazyShapeDtype.cpp and meta kernels.
+ * 
+ * It is only used by legacy ops that haven't been ported yet.
+ * */
+lazy_tensors::Shape InferShape(const torch::lazy::Node* node);
```

Comment:
1. Should we make InferShape also a TSNode method for each op?
2. A separate question, should we have LazyShapeDtype.cpp return just a Shape instead of the current two methods, size and dtype, given that we are going to keep the Shape class?

Reply:
For (2), yea, that's a great idea! 

For (1), do you mean making InferShape a class method of the TSNode class, instead of having standalone functions in the LazyShapeDtype file?  I think this could be nice; Currently we compute the shape first and feed it to the Node ctor; instead, we'd have to delay the initialization of the Shape field or use a static method instead of a class method.  What do you see as the main advantage of doing it this way?

Reply:
> For (1), do you mean making InferShape a class method of the TSNode class, instead of having standalone functions in the LazyShapeDtype file? I think this could be nice; Currently we compute the shape first and feed it to the Node ctor; instead, we'd have to delay the initialization of the Shape field or use a static method instead of a class method. What do you see as the main advantage of doing it this way?
The main advantage I am thinking is maybe this part of logic can also be codegen-ed, lik a simple calling into some meta method?

Reply:
Oh.  I think everything that can be codegenned already is codegenned.  We currently codegen either a call into meta kernel OR a call into handwritten inference functions in this part of codegen:
https://github.com/pytorch/pytorch/blob/lazy_tensor_staging/tools/codegen/dest/lazy_ir.py#L162

So I think changing it to a class method would just be a stylistic change, not a difference in terms of codegen.